lib/ex_unit/lib/ex_unit/on_exit_handler.ex

```diff
@@ -51,11 +73,29 @@ defmodule ExUnit.OnExitHandler do
 
   @spec run(pid, timeout) :: :ok | {Exception.kind(), term, Exception.stacktrace()}
   def run(pid, timeout) when is_pid(pid) do
-    [{^pid, sup, callbacks}] = :ets.take(@name, pid)
+    [{^pid, _manager, sup, callbacks}] = :ets.take(@name, pid)
     error = terminate_supervisor(sup, timeout)
     exec_on_exit_callbacks(Enum.reverse(callbacks), timeout, error)
   end
 
+  @spec start_failure_counter(manager) :: :ok
+  def start_failure_counter(manager) do
```

Comment:
Does the "failure counter" belong to the `ExUnit.OnExitHandler`, conceptually? (just to double check 🙂)

Reply:
It doesn't . I thought of that too. But thought creating another ETS for just a counter would be overkill.
let's ask what @josevalim  thinks about it

Reply:
probably we could implement it using an Agent